xxhash.h

```diff
@@ -813,8 +813,65 @@ XXH_PUBLIC_API XXH128_hash_t XXH128_hashFromCanonical(const XXH128_canonical_t*
  * routines for malloc() and free()
  */
 #include <stdlib.h>
-static void* XXH_malloc(size_t s) { return malloc(s); }
-static void  XXH_free  (void* p)  { free(p); }
+
+/*
+ * Malloc's a pointer that is always 64 byte aligned, to match the alignment
+ * of XXH3_state_t.
+ *
+ * This must be freed with `XXH_free()`.
+ *
+ * malloc typically guarantees 16 byte alignment on 64-bit systems and 8 byte
+ * alignment on 32-bit. This isn't enough for the 32 byte aligned loads in AVX2
+ * or on 32-bit, the 16 byte aligned loads in SSE2 and NEON.
+ *
+ * This underalignment previously caused a rather obvious crash which went
+ * completely unnoticed due to XXH3_createState() not actually being tested.
+ * Credit to RedSpah for noticing this bug.
+ *
+ * The alignment is done manually: Functions like posix_memalign or _mm_malloc
+ * are avoided: To maintain portability, we would have to write a fallback
+ * like this anyways, and besides, testing for the existence of library
+ * functions without relying on external build tools is impossible.
+ *
+ * The method is simple: Overallocate, manually align, and store the offset
+ * to the original behind the returned pointer.
+ */
+static void* XXH_malloc(size_t s)
+{
+    /* Overallocate to make room for manual realignment and an offset byte */
+    unsigned char* base = (unsigned char*)malloc(s + 64);
+    if (base != NULL) {
+        /*
+         * Get the offset needed to align this pointer to 64 bytes.
+         *
+         * Even if the returned pointer is 64-byte aligned, there will always
+         * be at least one byte to store the offset to the original pointer.
+         */
+        size_t offset = 64 - ((size_t)base % 64);
```

Comment:
It seems this will generalize 64-bytes alignment for all XXH states, including the XXH32 and XXH64 ones, which don't need it.

Maybe we could have an intermediate layer, like `XXH_malloc_align()`, that would be invoked by the XXH3 state constructor, while XXH32 and XXH64 continue to use the normal `XXH_malloc()`, which works well for them ?

Also, small detail, we can go the safe route to always align on 64-bytes, like the current static `struct` does,
or use the required alignment of the current target, which is supposed to be set by `XXH_ACC_ALIGN`, which would slightly reduce the amount of over-allocated memory.

Reply:
Not a bad idea.

I think we should generalize it to 32 bytes. Especially when we implement the dispatcher, we can't tell what the optimal alignment is. 

Unless we plan to add AVX512 (literally can't say it with a straight face 😛), 32 bytes is a good enough alignment.